utils/rjson.cc

```diff
@@ -556,4 +556,38 @@ sstring quote_json_string(const sstring& value) {
     return oss.str();
 }
 
+static future<> destroy_gently_nonleaf(rjson::value&& value_in) {
+    // We want the caller to move the value into this function, so 'value_in' is an rvalue reference.
+    // We want to hold the value while it's being destroyed, so we move it into the coroutine frame
+    // as the local 'value'.
+    auto value = std::move(value_in);
+
+    if (value.IsObject()) {
+        for (auto it = value.MemberBegin(); it != value.MemberEnd();) {
+            co_await destroy_gently(std::move(it->value));
+            it = value.EraseMember(it);
+        }
+    } else if (value.IsArray()) {
+        for (auto i = value.Size(); i > 0; --i) {
+            auto index = i - 1;
+            if (is_leaf(value[index])) {
```

Comment:
Why call destroy_gently if `value[index]` is a leaf, the latter will just return a ready future?
and what if it's not a leaf?

It looks like you meant the opposite logic: `if (!is_leaf(value[index]))` destroy_gently.
And value.Erase can probably be called unconditionally for both leaf and non-leaf entries.
And the the std::move to destroy_gently for the leaf case would not be needed.

Reply:
There's no need for is_leaf at all, it's handled by destroy_gently().